application/controllers/QuotasController.php

```diff
@@ -257,45 +249,34 @@ public function actionEditQuota($surveyid)
      */
     public function actionDeleteQuota()
     {
-        $surveyid = sanitize_int(Yii::app()->request->getPost('surveyid'));
-        if (!(Permission::model()->hasSurveyPermission($surveyid, 'quotas', 'delete'))) {
-            Yii::app()->user->setFlash('error', gT("Access denied."));
-            $this->redirect(Yii::app()->request->urlReferrer);
-        }
-
         $quotaId = Yii::app()->request->getPost('quota_id');
+        $oQuota = $this->getQuotaWithPermission($quotaId, 'delete');
+        $surveyid = $oQuota->sid;
+
         Quota::model()->deleteByPk($quotaId);
         QuotaLanguageSetting::model()->deleteAllByAttributes(array('quotals_quota_id' => $quotaId));
         QuotaMember::model()->deleteAllByAttributes(array('quota_id' => $quotaId));
 
         Yii::app()->user->setFlash('success', sprintf(gT("Quota with ID %s was deleted"), $quotaId));
 
-        $this->redirect($this->createUrl("quotas/index/surveyid/$surveyid"));
+        $this->redirect($this->createUrl(["quotas/index/",[ 'surveyid' => $surveyid]]));
```

Comment:
I don't know why, but this change forwards the app to public survey list index.php instead of quota admin. One customer get a white-age, dunno why yet.

Reply:
Customer get a 400 invalid request, actually.

Reply:
Yeah, it works when I restore this line. Funny.

Reply:
Maybe I'll just revert all those redirect changes...

Reply:
By comparing with other redirects, starting slash is missing here...

By looking here, there is some specific module behaviour if there is no starting slash
https://www.yiiframework.com/doc/api/1.1/CController#createUrl-detail

Reply:
:( sorry for mistake. Try ming remove laser / van eux issue.

Iis server ?

Reply:
@gabrieljenik nice, will have a look, thanks!

Reply:
Adding a slash did not help. Will just revert this line to its previous content.

Reply:
> Adding a slash did not help. Will just revert this line to its previous content.

It's not adding, it's remove … `$this->redirect($this->createUrl(["quotas/index",[ 'surveyid' => $surveyid]]));` 
no ?

Reply:
By looking at it again, I just feel the function is being called wrong.
Signature: `createUrl(string $route, array $params=array ( ), string $ampersand='&')`

We are only using the routes param if I am not mistaken, right?

Reply:
Right : must be `$this->redirect($this->createUrl("quotas/index",[ 'surveyid' => $surveyid]));`

Reply:
Olle fix it with `$this->redirect($this->createUrl("quotas/index/surveyid/{$surveyid}")`

(why it work when i test ????

Reply:
> (why it work when i test ????

You are sending the full route there, with the params embedded. Still, may not work for all route formats